java-checks/src/main/java/org/sonar/java/checks/helpers/TernaryValue.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the Sonar Source-Available License Version 1, as published by SonarSource SA.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the Sonar Source-Available License for more details.
+ *
+ * You should have received a copy of the Sonar Source-Available License
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
+package org.sonar.java.checks.helpers;
+
+import javax.annotation.Nullable;
+
+/** Represents the three possible values in three-valued logic. */
+public enum TernaryValue {
+  TRUE, FALSE, UNKNOWN;
+
+  public static TernaryValue ofBoolean(@Nullable Boolean b) {
+    if (b == null) {
+      return UNKNOWN;
+    }
+    return b ? TRUE : FALSE;
+  }
+
```

Comment:
I'm not sure about the usefulness of `ofBoolean` compared to `of`, I don't see where there could be ambiguity.
When we have a boolean like `isEnum()`, I would prefer having a dedicated method instead of converting it to `Boolean` in a method that converts it back to `boolean`.
```suggestion
  public static TernaryValue of(boolean b) {
    return b ? TRUE : FALSE;
  }

  public static TernaryValue of(@Nullable Boolean b) {
    return b == null ? UNKNOWN : of(b.booleanValue());
  }
```